fennel/client/client.py

```diff
@@ -409,42 +243,62 @@ def extract(
         return df
 
-    def extract_features(
-        self,
-        input_feature_list: List[Union[Feature, str]],
-        output_feature_list: List[Union[Feature, Featureset, str]],
-        input_dataframe: pd.DataFrame,
-        log: bool = False,
-        workflow: Optional[str] = None,
-        sampling_rate: Optional[float] = None,
-    ) -> Union[pd.DataFrame, pd.Series]:
+    # ----------------------- Branch API's -----------------------------------
+
+    def empty_branch(self, name: str):
         """
-        Going to be deprecated in favor of extract and will be removed in the future.
-        Extract features for a given output feature list from an input
-        feature list. The features are computed for the current time.
+        Create a new empty branch.
 
         Parameters:
-        input_feature_list (List[Union[Feature, str]]): List of feature objects or fully qualified feature names (when providing a str) can be used as input. We don't allow adding featureset as input because if an engineer adds a new feature to the featureset it would break all extract calls running in production.
-        output_feature_list (List[Union[Feature, Featureset, str]]): List of feature or featureset objects or fully qualified feature names (when providing a str) to compute.
-        input_dataframe (pd.DataFrame): Dataframe containing the input features.
-        log (bool): Boolean which indicates if the extracted features should also be logged (for log-and-wait approach to training data generation). Default is False.
-        workflow (Optional[str]): The name of the workflow associated with the feature extraction. Only relevant when log is set to True.
-        sampling_rate (float): The rate at which feature data should be sampled before logging. Only relevant when log is set to True. The default value is 1.0.
+        name (str): The name of the branch to create.
 
         Returns:
-        Union[pd.DataFrame, pd.Series]: Pandas dataframe or series containing the output features.
+        Dict[str, Any]: A dictionary containing the branch name and the status of the request.
         """
-        warnings.warn(
-            "This method is going to be deprecated in favor of extract."
-        )
-        return self.extract(
-            inputs=input_feature_list,
-            outputs=output_feature_list,
-            input_dataframe=input_dataframe,
-            log=log,
-            workflow=workflow,
-            sampling_rate=sampling_rate,
-        )
+        req = {"branch_name": name}
+        return self._post_json("{}/branch/create".format(V1_API), req)
+
+    def clone_branch(self, name: str, from_branch: str):
+        """
+        Clone a branch from another branch.
+
+        Parameters:
+        name (str): The name of the branch to create.
+        from_branch (str): The name of the branch to clone from.
+
+        Returns:
+        Dict[str, Any]: A dictionary containing the branch name and the status of the request.
```

Comment:
Why it is returning a dictionary? Do you imaging extending this method to clone many branches at once someday? If not, I'd imagine it should return just the `status` of that one branch that was requested?